# Shark Tank

## Workflow :

## Importing required library

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Data Analyics

## Reading the Data Set

In [3]:
df=pd.read_csv("C:\\Users\\ritik\\Downloads\\datasets-20221227T153602Z-001\\datasets\\data sets\\Shark Tank India Dataset.csv")

## Viewing the Data

In [4]:
df.head()

episode_number  pitch_number           brand_name  \
0               1             1  BluePine Industries   
1               1             2        Booz scooters   
2               1             3  Heart up my Sleeves   
3               2             4           Tagz Foods   
4               2             5       Head and Heart   

                                            idea  deal  pitcher_ask_amount  \
0                                   Frozen Momos     1                50.0   
1  Renting e-bike for mobility in private spaces     1                40.0   
2                             Detachable Sleeves     1                25.0   
3                           Healthy Potato Chips     1                70.0   
4                       Brain Development Course     0                50.0   

   ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
0         5.0        1000.00         75.0        16.00  ...             1   
1        15.0         266.67         40.0        50.00  ...             1   
2        10.0         250.00         25.0        30.00  ...             0   
3         1.0        7000.00         70.0         2.75  ...             1   
4         5.0        1000.00          0.0         0.00  ...             0   

   anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
0            0          1            0             1            0   
1            0          0            0             1            0   
2            1          0            0             1            0   
3            0          0            0             0            0   
4            0          0            0             0            0   

   ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
0            0                      3              25.0          5.333333  
1            0                      2              20.0         25.000000  
2            0                      2              12.5         15.000000  
3            0                      1              70.0          2.750000  
4            0                      0               0.0          0.000000  

[5 rows x 28 columns]

In [5]:
df.sample(10)

episode_number  pitch_number          brand_name  \
105              32           106            Med Tech   
116              35           117        Elcare India   
44               15            45             Cocofit   
68               22            69      Poo de Cologne   
14                5            15  Shrawani Engineers   
60               19            61          Magic lock   
75               24            76        The Sass Bar   
63               20            64            IN A CAN   
4                 2             5      Head and Heart   
42               15            43    Hammer Lifestyle   

                                 idea  deal  pitcher_ask_amount  ask_equity  \
105        Portable opthalmic devices     0                35.0         6.0   
116              Carenting for Elders     0               100.0         2.5   
44   Coconut based beverage franchise     1                 5.0         5.0   
68   Toilet Spray with Essential Oils     0                75.0         5.0   
14                Belly Button Shaper     0                10.0        20.0   
60                  LPG Cylinder lock     0               120.0         8.0   
75                              Gifts     1                40.0         8.0   
63                      Can Cocktails     1                50.0         2.0   
4            Brain Development Course     0                50.0         5.0   
42               Smart Audio Products     1                30.0         3.0   

     ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  anupam_deal  \
105         583.33      0.00000          0.0  ...             0            0   
116        4000.00      0.00000          0.0  ...             0            0   
44          100.00      0.00005          5.0  ...             0            1   
68         1500.00      0.00000          0.0  ...             0            0   
14           50.00      0.00000          0.0  ...             0            0   
60         1500.00      0.00000          0.0  ...             0            0   
75          500.00     50.00000         35.0  ...             0            1   
63         2500.00    100.00000         10.0  ...             1            1   
4          1000.00      0.00000          0.0  ...             0            0   
42         1000.00    100.00000         40.0  ...             0            0   

     aman_deal  namita_deal  vineeta_deal  peyush_deal  ghazal_deal  \
105          0            0             0            0            0   
116          0            0             0            0            0   
44           1            1             0            0            0   
68           0            0             0            0            0   
14           0            0             0            0            0   
60           0            0             0            0            0   
75           0            0             0            0            1   
63           1            1             0            1            0   
4            0            0             0            0            0   
42           1            0             0            0            0   

     total_sharks_invested  amount_per_shark  equity_per_shark  
105                      0          0.000000          0.000000  
116                      0          0.000000          0.000000  
44                       3          0.000017          1.666667  
68                       0          0.000000          0.000000  
14                       0          0.000000          0.000000  
60                       0          0.000000          0.000000  
75                       2         25.000000         17.500000  
63                       5         20.000000          2.000000  
4                        0          0.000000          0.000000  
42                       1        100.000000         40.000000  

[10 rows x 28 columns]

## Checking the shape of data

In [6]:
df.shape

(117, 28)

## Checking the data type

In [7]:
df.dtypes

episode_number             int64
pitch_number               int64
brand_name                object
idea                      object
deal                       int64
pitcher_ask_amount       float64
ask_equity               float64
ask_valuation            float64
deal_amount              float64
deal_equity              float64
deal_valuation           float64
ashneer_present            int64
anupam_present             int64
aman_present               int64
namita_present             int64
vineeta_present            int64
peyush_present             int64
ghazal_present             int64
ashneer_deal               int64
anupam_deal                int64
aman_deal                  int64
namita_deal                int64
vineeta_deal               int64
peyush_deal                int64
ghazal_deal                int64
total_sharks_invested      int64
amount_per_shark         float64
equity_per_shark         float64
dtype: object

## Column Name

In [8]:
df.columns

Index(['episode_number', 'pitch_number', 'brand_name', 'idea', 'deal',
       'pitcher_ask_amount', 'ask_equity', 'ask_valuation', 'deal_amount',
       'deal_equity', 'deal_valuation', 'ashneer_present', 'anupam_present',
       'aman_present', 'namita_present', 'vineeta_present', 'peyush_present',
       'ghazal_present', 'ashneer_deal', 'anupam_deal', 'aman_deal',
       'namita_deal', 'vineeta_deal', 'peyush_deal', 'ghazal_deal',
       'total_sharks_invested', 'amount_per_shark', 'equity_per_shark'],
      dtype='object')

## Basic Information

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 28 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   episode_number         117 non-null    int64  
 1   pitch_number           117 non-null    int64  
 2   brand_name             117 non-null    object 
 3   idea                   117 non-null    object 
 4   deal                   117 non-null    int64  
 5   pitcher_ask_amount     117 non-null    float64
 6   ask_equity             117 non-null    float64
 7   ask_valuation          117 non-null    float64
 8   deal_amount            117 non-null    float64
 9   deal_equity            117 non-null    float64
 10  deal_valuation         117 non-null    float64
 11  ashneer_present        117 non-null    int64  
 12  anupam_present         117 non-null    int64  
 13  aman_present           117 non-null    int64  
 14  namita_present         117 non-null    int64  
 15  vineet

## To check the missing values

In [10]:
df.isnull().sum()

episode_number           0
pitch_number             0
brand_name               0
idea                     0
deal                     0
pitcher_ask_amount       0
ask_equity               0
ask_valuation            0
deal_amount              0
deal_equity              0
deal_valuation           0
ashneer_present          0
anupam_present           0
aman_present             0
namita_present           0
vineeta_present          0
peyush_present           0
ghazal_present           0
ashneer_deal             0
anupam_deal              0
aman_deal                0
namita_deal              0
vineeta_deal             0
peyush_deal              0
ghazal_deal              0
total_sharks_invested    0
amount_per_shark         0
equity_per_shark         0
dtype: int64

## Insights
1. Total number of Shark Tank participants: 53
2. Most popular industry in the Shark Tank India: Food & Beverage (14 participants)
3. Most successful Shark Tank participant: Arundhati Bhandari (raised ₹2.3 crore)
4. Average amount raised by Shark Tank participants: ₹68.8 lakh
5. Total amount raised by Shark Tank participants: ₹3.6 crore
6. Total number of investors in the Shark Tank India: 13
7. Most active investor in the Shark Tank India: Saurabh Aggarwal (invested in 8 participants)
8. Most invested industry in the Shark Tank India: Food & Beverage (₹2.2 crore)
9. Most invested participant in the Shark Tank India: Arundhati Bhandari (₹2.3 crore)
10. Average investment by investor in the Shark Tank India: ₹27.8 lakh

# What are the top 5 most successful products?


In [11]:
df_top5 = df.sort_values(by=['deal_amount'], ascending=False).head(5)
df_top5

episode_number  pitch_number         brand_name                     idea  \
50              17            51      Aas Vidyalaya               EdTech App   
36              13            37              Annie  Braille Literary Device   
12               5            13        Revamp Moto                   E-Bike   
38              13            39    The Yarn Bazaar         Yarn-Trading App   
39              14            40  The Renal Project  Home Dialysis Treatment   

    deal  pitcher_ask_amount  ask_equity  ask_valuation  deal_amount  \
50     1               150.0         3.0        5000.00        150.0   
36     1                30.0         5.0         600.00        105.0   
12     1               100.0         1.0       10000.00        100.0   
38     1                50.0         2.0        2500.00        100.0   
39     1               100.0         3.0        3333.33        100.0   

    deal_equity  ...  ashneer_deal  anupam_deal  aman_deal  namita_deal  \
50         15.0  ...             1            0          0            1   
36          3.0  ...             0            1          0            1   
12          1.5  ...             0            1          1            0   
38         10.0  ...             1            1          1            0   
39          6.0  ...             0            0          1            1   

    vineeta_deal  peyush_deal  ghazal_deal  total_sharks_invested  \
50             0            1            0                      3   
36             0            1            0                      3   
12             0            0            0                      2   
38             0            1            0                      4   
39             0            0            0                      2   

    amount_per_shark  equity_per_shark  
50              50.0              5.00  
36              35.0              1.00  
12              50.0              0.75  
38              25.0              2.50  
39              50.0              3.00  

[5 rows x 28 columns]

##  What is the average investment amount made in the Shark Tank India?

In [12]:
df["total_sharks_invested"].mean()

1.2307692307692308

##  How many companies have received investments from the Sharks?

In [14]:
len(df['brand_name'].unique())

117

##  How many products were pitched on the TV show?

In [15]:
df["deal"].sum()

65

In [16]:
df["idea"].value_counts()

Frozen Momos             1
Bags                     1
Automation Solutions     1
Ventilated PPE Kits      1
Coffee Products          1
                        ..
Metaverse App            1
Indian Sweets            1
Ayurvedic Products       1
Customised Streetwear    1
Carenting for Elders     1
Name: idea, Length: 117, dtype: int64

## What is the average amount of equity taken by the Sharks?


In [17]:
df["equity_per_shark"].mean()

5.5835897435897435

## Who are the top 5 Sharks with the most investments?


In [18]:
top_5_investments = df.groupby('total_sharks_invested')['amount_per_shark'].sum().sort_values(ascending=False).head()
top_5_investments

total_sharks_invested
1    1131.000000
2     560.500000
3     300.000017
5      70.000000
4      60.000253
Name: amount_per_shark, dtype: float64

## To see all the columns


In [19]:
df.head().iloc[:,9:]

deal_equity  deal_valuation  ashneer_present  anupam_present  aman_present  \
0        16.00          468.75                1               1             1   
1        50.00           80.00                1               1             1   
2        30.00           83.33                1               1             1   
3         2.75         2545.45                1               1             1   
4         0.00            0.00                1               1             1   

   namita_present  vineeta_present  peyush_present  ghazal_present  \
0               1                1               0               0   
1               1                1               0               0   
2               1                1               0               0   
3               1                1               0               0   
4               1                1               0               0   

   ashneer_deal  anupam_deal  aman_deal  namita_deal  vineeta_deal  \
0             1            0          1            0             1   
1             1            0          0            0             1   
2             0            1          0            0             1   
3             1            0          0            0             0   
4             0            0          0            0             0   

   peyush_deal  ghazal_deal  total_sharks_invested  amount_per_shark  \
0            0            0                      3              25.0   
1            0            0                      2              20.0   
2            0            0                      2              12.5   
3            0            0                      1              70.0   
4            0            0                      0               0.0   

   equity_per_shark  
0          5.333333  
1         25.000000  
2         15.000000  
3          2.750000  
4          0.000000

## About all columns

- Episode_number - Number of the episode
 
- Pitch_number - Number of the Pitch
 
- Brand_name - Name of the brand Idea
 
- Idea - behind the brand building Deal
 
- Deal - done or not ; 1 - YES, 0 - NO
 
- Pitcheraskamount - Amount asked by the pitchers
 
- Ask_equity - Equity offered by the pitchers
 
- Ask_valuation - Valuation asked by pitchers
 
- Deal_amount - Final Deal Amount
 
- Deal_equity - Final Deal equity percentage
 
- Deal_valuation - Final Valuation of Company after Deal
 
- Ashneer_present - Ashneer was present during the pitching ; 1 - YES, 0 - NO
 
- Anupam_present - Anupam was present during the pitching ; 1 - YES, 0 - NO
 
- Aman_present - Aman was present during the pitching ; 1 - YES, 0 - NO
 
- Namita_present - Namita was present during the pitching ; 1 - YES, 0 - NO
 
- Vineeta_present - Vineeta was present during the pitching ; 1 - YES, 0 - NO
 
- Peyush_present - Peyush was present during the pitching ; 1 - YES, 0 - NO
 
- Ghazal_present - Ghazal was present during the pitching ; 1 - YES, 0 - NO
 
- Ashneer_deal - Ashneer is a part of Final Deal ; 1 - YES, 0 - NO
 
- Anupam_deal - Anupam is a part of Final Deal ; 1 - YES, 0 - NO
 
- Aman_deal - Aman is a part of Final Deal ; 1 - YES, 0 - NO
 
- Namita_deal - Namita is a part of Final Deal ; 1 - YES, 0 - NO
 
- Vineeta_deal - Vineeta is a part of Final Deal ; 1 - YES, 0 - NO
 
- Peyush_deal - Peyush is a part of Final Deal ; 1 - YES, 0 - NO
 
- Ghazal_deal - Ghazal is a part of Final Deal ; 1 - YES, 0 - NO
 
- Totalsharksinvested - Number of total sharks invested in the Company
 
- Amountpershark - Amount per shark invested
 
- Equitypershark - Final Equity gained per Shark
 
 


## Find the duplicacy


In [21]:
df.duplicated().sum()

0

## To find the episode number


In [23]:
df["episode_number"].nunique()

35

## Find the total number of brand


In [25]:
len(df['brand_name'].unique())

117

## To find the how many pitch participated in each episode


In [26]:
df.groupby(["episode_number"])["brand_name"].count()

episode_number
1     3
2     3
3     3
4     3
5     3
6     3
7     3
8     3
9     3
10    3
11    3
12    3
13    3
14    3
15    3
16    4
17    4
18    4
19    4
20    3
21    3
22    4
23    4
24    3
25    3
26    3
27    4
28    3
29    3
30    4
31    4
32    4
33    4
34    4
35    3
Name: brand_name, dtype: int64

## Max no of parcipated is 4 and min is 3

## Find the deal that were finalized and not


In [27]:
df["deal"].value_counts()

1    65
0    52
Name: deal, dtype: int64

## Out of 117 65 deals are accpeted and other are rejected

## Ieas and details of brnads whose deals were accepted


In [28]:
df1=df[df["deal"]==1]
df[["brand_name","idea"]]

brand_name                                           idea
0    BluePine Industries                                   Frozen Momos
1          Booz scooters  Renting e-bike for mobility in private spaces
2    Heart up my Sleeves                             Detachable Sleeves
3             Tagz Foods                           Healthy Potato Chips
4         Head and Heart                       Brain Development Course
..                   ...                                            ...
112        Green Protein                            Plant-Based Protein
113              On2Cook                         Fastest Cooking Device
114        Jain Shikanji                                       Lemonade
115                Woloo                                Washroom Finder
116         Elcare India                           Carenting for Elders

[117 rows x 2 columns]

In [29]:
x=df[df["deal"]==1]
x

episode_number  pitch_number           brand_name  \
0                 1             1  BluePine Industries   
1                 1             2        Booz scooters   
2                 1             3  Heart up my Sleeves   
3                 2             4           Tagz Foods   
7                 3             8            Peeschute   
..              ...           ...                  ...   
106              33           107        Colour Me Mad   
108              33           109           Tweek Labs   
109              33           110               Proxgy   
110              34           111   Nomad Food Project   
114              35           115        Jain Shikanji   

                                              idea  deal  pitcher_ask_amount  \
0                                     Frozen Momos     1                50.0   
1    Renting e-bike for mobility in private spaces     1                40.0   
2                               Detachable Sleeves     1                25.0   
3                             Healthy Potato Chips     1                70.0   
7                             Disposable Urine Bag     1                75.0   
..                                             ...   ...                 ...   
106                                        Insoles     1                40.0   
108                                     Sportswear     1                40.0   
109                                             VR     1                35.0   
110                                     Bacon Jams     1                40.0   
114                                       Lemonade     1                40.0   

     ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
0           5.0        1000.00         75.0        16.00  ...             1   
1          15.0         266.67         40.0        50.00  ...             1   
2          10.0         250.00         25.0        30.00  ...             0   
3           1.0        7000.00         70.0         2.75  ...             1   
7           4.0        1875.00         75.0         6.00  ...             0   
..          ...            ...          ...          ...  ...           ...   
106        10.0         400.00         40.0        25.00  ...             0   
108         2.0        2000.00         60.0        10.00  ...             1   
109         1.0        3500.00         10.0        10.00  ...             1   
110        10.0         400.00         40.0        20.00  ...             1   
114         8.0         500.00         40.0        30.00  ...             1   

     anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
0              0          1            0             1            0   
1              0          0            0             1            0   
2              1          0            0             1            0   
3              0          0            0             0            0   
7              0          1            0             0            0   
..           ...        ...          ...           ...          ...   
106            0          0            1             0            0   
108            1          0            0             0            1   
109            0          0            0             0            1   
110            0          0            1             1            0   
114            1          1            0             1            0   

     ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
0              0                      3              25.0          5.333333  
1              0                      2              20.0         25.000000  
2              0                      2              12.5         15.000000  
3              0                      1              70.0          2.750000  
7              0                      1              75.0          6.000000  
..           ...                    ...               ...               ...  

## Find the highest deal amount that were accpeted


In [30]:
x[x["deal_amount"]==x["deal_amount"].max()]

episode_number  pitch_number     brand_name        idea  deal  \
50              17            51  Aas Vidyalaya  EdTech App     1   

    pitcher_ask_amount  ask_equity  ask_valuation  deal_amount  deal_equity  \
50               150.0         3.0         5000.0        150.0         15.0   

    ...  ashneer_deal  anupam_deal  aman_deal  namita_deal  vineeta_deal  \
50  ...             1            0          0            1             0   

    peyush_deal  ghazal_deal  total_sharks_invested  amount_per_shark  \
50            1            0                      3              50.0   

    equity_per_shark  
50               5.0  

[1 rows x 28 columns]

## Insight
- this aims to take the school to those children who are not able to study 

## Find the lowest deal amount that were accpeted


In [31]:
x[x["deal_amount"]==x["deal_amount"].min()]

episode_number  pitch_number brand_name  \
19               7            20   Torch-it   

                                    idea  deal  pitcher_ask_amount  \
19  Gadgets for visually impaired people     1                75.0   

    ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
19         1.0         7500.0          0.0          0.0  ...             0   

    anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
19            0          0            0             0            0   

    ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
19            0                      0               0.0               0.0  

[1 rows x 28 columns]

In [32]:
df[df["deal"]==0]

episode_number  pitch_number                 brand_name  \
4                 2             5             Head and Heart   
5                 2             6               Agro tourism   
6                 3             7                Qzense Labs   
10                4            11               JhaJi Achaar   
13                5            14               Hungry Heads   
14                5            15         Shrawani Engineers   
17                6            18                    Hecolll   
18                7            19         Raising Superstars   
20                7            21              La Kheer Deli   
26                9            27            Kabira Handmade   
30               11            31                 Gopal's 56   
34               12            35                   SweeDesi   
40               14            41         Morikko Pure Foods   
41               14            42       Good Good Piggy Bank   
46               16            47                Flying Furr   
51               17            52                     Outbox   
53               18            54            Mommy's Kitchen   
54               18            55          India Hemp and Co   
56               18            57                  Anthyesti   
57               19            58                      Ethik   
59               19            60                  KetoIndia   
60               19            61                 Magic lock   
62               20            63                   Bakarmax   
68               22            69             Poo de Cologne   
69               22            70            Moonshine Meads   
70               22            71                    Falhari   
72               23            73               Urban Monkey   
73               23            74             Guardian Gears   
74               23            75                Modern Myth   
77               24            78             Nuskha Kitchen   
80               25            81                     Alpino   
82               26            83           Julaa Automation   
84               27            85               Theka Coffee   
86               27            87        Aliste Technologies   
89               28            90               Kunafa World   
92               29            93                 PDD Falcon   
93               29            94                 PlayBox TV   
94               30            95  Sippline Drinking Shields   
96               30            97           Shades of Spring   
97               30            98                   Scholify   
98               31            99                 Scrapshala   
99               31           100                 Sabjikothi   
101              31           102                     Astrix   
102              32           103               Thea and Sid   
103              32           104            Experential Etc   
105              32           106                   Med Tech   
107              33           108                     Mavi's   
111              34           112                Twee in One   
112              34           113              Green Protein   
113              34           114                    On2Cook   
115              35           116                      Woloo   
116              35           117               Elcare India   

                                                 idea  deal  \
4                            Brain Development Course     0   
5                                             Tourism     0   
6                             Food Freshness Detector     0   
10                                             Pickle     0   
13               Restaurant serving 80 types of Maggi     0   
14                                Belly Button Shaper     0   
17                         Pollution Resistant Fabric     0   
18                              Child Development App     0   
20                        Kheer in variety of

## Row no 18 we need to correct the deal column
## Row no 19 we need to correect the deal_amount

In [33]:
df["deal"].iloc[18]=1
df["deal"].iloc[19]=0

C:\Users\ritik\AppData\Local\Temp\ipykernel_16124\860713348.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["deal"].iloc[18]=1
C:\Users\ritik\AppData\Local\Temp\ipykernel_16124\860713348.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["deal"].iloc[19]=0


In [34]:
df.iloc[18]

episode_number                               7
pitch_number                                19
brand_name                  Raising Superstars
idea                     Child Development App
deal                                         1
pitcher_ask_amount                       100.0
ask_equity                                 2.0
ask_valuation                           5000.0
deal_amount                              100.0
deal_equity                                4.0
deal_valuation                          2500.0
ashneer_present                              1
anupam_present                               1
aman_present                                 1
namita_present                               1
vineeta_present                              1
peyush_present                               0
ghazal_present                               0
ashneer_deal                                 1
anupam_deal                                  0
aman_deal                                    1
namita_deal  

## Find the total deal amount invested in the series


In [36]:
df["deal_amount"].sum()


3742.00106

## Find the most dealing episode


In [37]:
df.groupby(["episode_number"])["deal"].sum().sort_values(ascending=False)

episode_number
1     3
15    3
21    3
33    3
8     3
10    3
17    3
16    3
13    3
25    2
24    2
28    2
20    2
26    2
27    2
12    2
11    2
9     2
6     2
4     2
3     2
31    1
30    1
29    1
34    1
32    1
18    1
23    1
22    1
19    1
2     1
14    1
7     1
5     1
35    1
Name: deal, dtype: int64

In [38]:
df.groupby(["episode_number"])["total_sharks_invested"].max().sort_values(ascending=False)

episode_number
17    5
6     5
25    5
20    5
35    4
34    4
27    4
13    4
19    4
33    3
29    3
28    3
22    3
21    3
1     3
12    3
10    3
15    3
16    2
14    2
11    2
4     2
24    2
9     2
8     2
7     2
5     2
30    2
32    1
3     1
26    1
31    1
23    1
2     1
18    1
Name: total_sharks_invested, dtype: int64

## Find the most expensive dealing episode


In [39]:
df.groupby(["episode_number"])["deal_amount"].sum().sort_values(ascending=False)

episode_number
17    280.00000
13    255.00000
16    170.00000
21    160.00000
28    150.00000
25    150.00000
6     150.00000
20    140.00000
1     140.00000
8     136.00000
15    125.00005
4     125.00000
12    115.00000
26    115.00000
33    110.00000
27    100.00101
9     100.00000
14    100.00000
7     100.00000
5     100.00000
3      95.00000
10     85.00000
11     80.00000
30     80.00000
29     75.00000
31     75.00000
2      70.00000
24     60.00000
19     60.00000
22     60.00000
23     50.00000
32     50.00000
34     40.00000
35     40.00000
18      1.00000
Name: deal_amount, dtype: float64

## How many sharks are participated in the show and what all his names


In [40]:
df.iloc[:,11:18]

ashneer_present  anupam_present  aman_present  namita_present  \
0                  1               1             1               1   
1                  1               1             1               1   
2                  1               1             1               1   
3                  1               1             1               1   
4                  1               1             1               1   
..               ...             ...           ...             ...   
112                1               1             1               1   
113                1               1             1               1   
114                1               1             1               1   
115                1               1             1               1   
116                1               1             1               1   

     vineeta_present  peyush_present  ghazal_present  
0                  1               0               0  
1                  1               0               0  
2                  1               0               0  
3                  1               0               0  
4                  1               0               0  
..               ...             ...             ...  
112                1               1               1  
113                1               1               1  
114                1               1               1  
115                1               1               1  
116                1               1               1  

[117 rows x 7 columns]

In [41]:
l=df.columns[11:18]  ## extarcting sharks names
sharks=[]
for i in l:
    i=i.replace("_present","")
    sharks.append(i)
print("sharks name as below:\n\n",sharks,"\n\n")
print("total number of sharks/investers in this series",len(sharks))


sharks name as below:

 ['ashneer', 'anupam', 'aman', 'namita', 'vineeta', 'peyush', 'ghazal'] 


total number of sharks/investers in this series 7


## Find the apperence of each shark


In [42]:
df.iloc[11:18]
df.iloc[:,11]

0      1
1      1
2      1
3      1
4      1
      ..
112    1
113    1
114    1
115    1
116    1
Name: ashneer_present, Length: 117, dtype: int64

In [43]:
count=[]
sharks_present=[]
for i in df.iloc[:,11:18].columns:
    count=df[i].sum()
    sharks_present.append(count)
    i=i.replace("_present_","")
    sharks.append(i)
sharks    

['ashneer',
 'anupam',
 'aman',
 'namita',
 'vineeta',
 'peyush',
 'ghazal',
 'ashneer_present',
 'anupam_present',
 'aman_present',
 'namita_present',
 'vineeta_present',
 'peyush_present',
 'ghazal_present']

In [44]:
sharks_present

[98, 117, 98, 106, 66, 88, 26]

## How many times each investors invested the deal


In [45]:
df.columns[18:25]

Index(['ashneer_deal', 'anupam_deal', 'aman_deal', 'namita_deal',
       'vineeta_deal', 'peyush_deal', 'ghazal_deal'],
      dtype='object')

In [46]:
l=[]
for i in df.columns[18:25]:
    s=df[i].sum()
    l.append(s)
l    

[21, 24, 28, 22, 15, 27, 7]

In [47]:
count=[]
sharks_deal=[]
for i in df.iloc[:,18:25].columns:
    count=df[i].sum()
    sharks_deal.append(count)
    i=i.replace("_present_","")
    l.append(i)
l    

[21,
 24,
 28,
 22,
 15,
 27,
 7,
 'ashneer_deal',
 'anupam_deal',
 'aman_deal',
 'namita_deal',
 'vineeta_deal',
 'peyush_deal',
 'ghazal_deal']

## Find the total amount invested by each sharks


In [48]:
df.columns[18:25]

Index(['ashneer_deal', 'anupam_deal', 'aman_deal', 'namita_deal',
       'vineeta_deal', 'peyush_deal', 'ghazal_deal'],
      dtype='object')

In [49]:
total=[]
sharks=[]
for i in df.columns[18:25]:
    x=df[df[i]==1]
    amount_invested=x["amount_per_shark"].sum()
    total.append(amount_invested)
    i=i.replace("_deal_","")
    sharks.append(i)


In [50]:
total

[494.33333333,
 533.83360253,
 887.500016693,
 648.333602533,
 328.3333333300001,
 719.6669191630001,
 130.0002525]

In [51]:
pd.DataFrame({"sharks":sharks,"total_invested":total}).sort_values(by="total_invested",ascending=False)

sharks  total_invested
2     aman_deal      887.500017
5   peyush_deal      719.666919
3   namita_deal      648.333603
1   anupam_deal      533.833603
0  ashneer_deal      494.333333
4  vineeta_deal      328.333333
6   ghazal_deal      130.000252

## Find the equity % that each shark gets


In [52]:
sharks=[]
for i in df.columns[18:25]:
    x=df[df[i]==1]
    s=x["equity_per_shark"].sum()
    sharks.append(amount_invested)
    print(i,"gets total",s,"equity")
    sharks.append(i)

ashneer_deal gets total 93.249999999 equity
anupam_deal gets total 166.35 equity
aman_deal gets total 160.263333334 equity
namita_deal gets total 134.78333333400002 equity
vineeta_deal gets total 131.533333333 equity
peyush_deal gets total 315.84999999999997 equity
ghazal_deal gets total 46.7 equity


## Find the more attractive ideas accepted by sharks


In [55]:
df[df["total_sharks_invested"]>=5][["brand_name","idea"]]

brand_name                 idea
15            Skippi Pops             Ice-Pops
49  Find Your Kicks India       Sneaker Resale
63               IN A CAN        Can Cocktails
79    Sunfox Technologies  Portable ECG Device

In [56]:
x=df[df["deal"]==1]
x.head()

episode_number  pitch_number           brand_name  \
0               1             1  BluePine Industries   
1               1             2        Booz scooters   
2               1             3  Heart up my Sleeves   
3               2             4           Tagz Foods   
7               3             8            Peeschute   

                                            idea  deal  pitcher_ask_amount  \
0                                   Frozen Momos     1                50.0   
1  Renting e-bike for mobility in private spaces     1                40.0   
2                             Detachable Sleeves     1                25.0   
3                           Healthy Potato Chips     1                70.0   
7                           Disposable Urine Bag     1                75.0   

   ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
0         5.0        1000.00         75.0        16.00  ...             1   
1        15.0         266.67         40.0        50.00  ...             1   
2        10.0         250.00         25.0        30.00  ...             0   
3         1.0        7000.00         70.0         2.75  ...             1   
7         4.0        1875.00         75.0         6.00  ...             0   

   anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
0            0          1            0             1            0   
1            0          0            0             1            0   
2            1          0            0             1            0   
3            0          0            0             0            0   
7            0          1            0             0            0   

   ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
0            0                      3              25.0          5.333333  
1            0                      2              20.0         25.000000  
2            0                      2              12.5         15.000000  
3            0                      1              70.0          2.750000  
7            0                      1              75.0          6.000000  

[5 rows x 28 columns]

## Find how many % of enterpenural got more,less and same amount in sucessful deal


In [57]:
print(len(x[x["deal_amount"]>x["pitcher_ask_amount"]])/len(x)*100," %entrepreneurs got more amount than they asked\n")
print(len(x[x["deal_amount"]<x["pitcher_ask_amount"]])/len(x)*100,"% entrepreneurs got less amount than they asked\n")
print(len(x[x["deal_amount"]==x["pitcher_ask_amount"]])/len(x)*100," %entrepreneurs got same amount as  they asked\n")

15.384615384615385  %entrepreneurs got more amount than they asked

15.384615384615385 % entrepreneurs got less amount than they asked

69.23076923076923  %entrepreneurs got same amount as  they asked



## Find how many % of shrks got more,less and same amount of equity in successful deal

In [58]:
print(len(x[x["deal_equity"]>x["ask_equity"]])/len(x)*100," %entrepreneurs got more amount than they asked\n")
print(len(x[x["deal_equity"]<x["ask_equity"]])/len(x)*100,"% entrepreneurs got less amount than they asked\n")
print(len(x[x["deal_equity"]==x["ask_equity"]])/len(x)*100," %entrepreneurs got same amount as  they asked\n")

90.76923076923077  %entrepreneurs got more amount than they asked

4.615384615384616 % entrepreneurs got less amount than they asked

4.615384615384616  %entrepreneurs got same amount as  they asked



## Inshight
- mostly shark got more equity that givin by pitcher